src/qt/walletcontroller.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <qt/walletcontroller.h>
+
+#include <interfaces/handler.h>
+#include <interfaces/node.h>
+#include <QMutexLocker>
+#include <QThread>
+
+WalletController::WalletController(interfaces::Node& node, const PlatformStyle* platform_style, OptionsModel* options_model, QObject* parent)
+    : QObject(parent)
+    , m_node(node)
+    , m_platform_style(platform_style)
+    , m_options_model(options_model)
+{
+    m_handler_load_wallet = m_node.handleLoadWallet([this](std::unique_ptr<interfaces::Wallet> wallet) {
+        getOrCreateModel(std::move(wallet));
+    });
+
+    for (auto& wallet : m_node.getWallets()) {
+        getOrCreateModel(std::move(wallet));
+    }
+}
+
+// Not using the default destructor because not all member types definitions are
+// available in the header, just forward declared.
+WalletController::~WalletController() {}
+
+std::vector<WalletModel*> WalletController::getWalletModels() const
+{
+    QMutexLocker locker(&m_mutex);
+    std::vector<WalletModel*> wallet_models;
+    wallet_models.reserve(m_wallet_models.size());
+    for (auto it : m_wallet_models) {
+        wallet_models.push_back(it.second);
+    }
+    return wallet_models;
+}
+
+std::vector<std::string> WalletController::getWalletsAvailableToLoad() const
```

Comment:
Is this unused?

Reply:
Used in #15153, I can move there if you prefer?

Reply:
I'd recommend moving it there. I think it's best not to have unused code in PRs.